sky/sdk/lib/widgets/block_viewport.dart

```diff
@@ -26,16 +19,71 @@ class _Key {
   final String key;
   bool operator ==(other) => other is _Key && other.type == type && other.key == key;
   int get hashCode => 373 * 37 * type.hashCode + key.hashCode;
+  String toString() => "_Key(type: $type, key: $key)";
+}
+
+typedef void LayoutChangedCallback();
+
+class BlockViewportLayoutState {
+  BlockViewportLayoutState() :
```

Comment:
We've been putting the : on the next line, indented two spaces from the previous line.